core/keyboard_nav/key_map.js

```diff
@@ -0,0 +1,250 @@
+  });
+
+  //Set A
+  var aAction = new Blockly.KeyAction('out', 'Goes out', function() {
+    if (Blockly.Navigation.currentState_ === Blockly.Navigation.STATE_WS) {
+      Blockly.Navigation.cursor_.out();
+    } else if (Blockly.Navigation.currentState_ === Blockly.Navigation.STATE_FLYOUT) {
+      Blockly.Navigation.focusToolbox();
+    } else if (Blockly.Navigation.currentState_ === Blockly.Navigation.STATE_TOOLBOX) {
+      Blockly.Navigation.outCategory();
+    }
+  });
+
+  //Set S
+  var sAction = new Blockly.KeyAction('next', 'Goes to the next location', function() {
+    if (Blockly.Navigation.currentState_ === Blockly.Navigation.STATE_WS) {
+      Blockly.Navigation.cursor_.next();
+    } else if (Blockly.Navigation.currentState_ === Blockly.Navigation.STATE_FLYOUT) {
+      Blockly.Navigation.selectNextBlockInFlyout();
+    } else if (Blockly.Navigation.currentState_ === Blockly.Navigation.STATE_TOOLBOX) {
+      Blockly.Navigation.nextCategory();
+    }
+  });
+
+  //Set D
+  var dAction = new Blockly.KeyAction('in', 'Goes in', function() {
+    if (Blockly.Navigation.currentState_ === Blockly.Navigation.STATE_WS) {
+      Blockly.Navigation.cursor_.in();
+    } else if (Blockly.Navigation.currentState_ === Blockly.Navigation.STATE_TOOLBOX) {
+      Blockly.Navigation.inCategory();
+    }
+  });
+
+  //Set I
+  var iAction = new Blockly.KeyAction('insert',
+      'Tries to connect the current location to the marked location', function() {
+        if (Blockly.Navigation.currentState_ === Blockly.Navigation.STATE_WS) {
+          Blockly.Navigation.modify();
+        }
+      });
+
+  //Set Enter
+  var enterAction = new Blockly.KeyAction('mark', 'Marks the current location', function() {
+    if (Blockly.Navigation.currentState_ === Blockly.Navigation.STATE_WS) {
+      Blockly.Navigation.handleEnterForWS();
+    } else if (Blockly.Navigation.currentState_ === Blockly.Navigation.STATE_FLYOUT) {
+      Blockly.Navigation.insertFromFlyout();
+    }
+  });
+
+  //Set X
+  var xAction = new Blockly.KeyAction('disconnect', 'Disconnect the blocks', function() {
+    if (Blockly.Navigation.currentState_ === Blockly.Navigation.STATE_WS) {
+      Blockly.Navigation.disconnectBlocks();
+    }
+  });
+
+  //Set T
+  var tAction = new Blockly.KeyAction('toolbox', 'Open the toolbox', function() {
+    if (!Blockly.getMainWorkspace().getToolbox()) {
+      Blockly.Navigation.focusFlyout();
+    } else {
+      Blockly.Navigation.focusToolbox();
+    }
+  });
+
+  //Set E
+  var eAction = new Blockly.KeyAction('exit', 'Exit the toolbox', function() {
+    if (Blockly.Navigation.currentState_ === Blockly.Navigation.STATE_TOOLBOX) {
+      Blockly.Navigation.focusWorkspace();
+    }
+  });
+
+  //Set ESC
+  var escAction = new Blockly.KeyAction('escape', 'Exit the toolbox', function() {
+    if (Blockly.Navigation.currentState_ === Blockly.Navigation.STATE_TOOLBOX) {
+      Blockly.Navigation.focusWorkspace();
+    }
+  });
+
+  var map = {};
```

Comment:
Building this map is the only thing this function should do, using publicly defined actions.

Reply:
done

Reply:
done